### ISSUE TITLE
Watcher for Quarkus

### ISSUE THREAD

**Issue by devang-gaur** [eclipse-jkube/jkube#256]
Hi, although I don't have experience using Quarkus, I do know that It supports watch mode (or dev mode) out of the box, just like spring. So, we should add a Watcher for Quarkus, on the lines of the Spring-boot Watcher we have.

**Comment by manusa** [comment]
Having a specific Quarkus watch goal sounds very good. However I'm not sure if your statement "It supports watch mode (or dev mode) out of the box, just like spring" may be based in a wrong conclusion. 

If you are referring to `quarkus:dev` Maven goal, this is sort of the equivalent to `spring-boot:run` which is **not** what we are using in [SpringBootWatcher](https://github.com/eclipse/jkube/blob/master/jkube-kit/jkube-kit-spring-boot/src/main/java/org/eclipse/jkube/springboot/watcher/SpringBootWatcher.java). This takes care of hot reloading the application in the local machine.

For Spring we are using the [`RemoteSpringApplication`](https://docs.spring.io/autorepo/docs/spring-boot/1.4.5.RELEASE/reference/html/using-boot-devtools.html#using-boot-devtools-remote) which takes care of reloading in the **remote** java application whatever is changed in the **local** classpath or defined file watch list.

Having said all this I see that Quarkus has a `[remote-dev](https://quarkus.io/guides/maven-tooling#remote-development-mode)` mode (maybe you are referring to this instead). This sounds definitely something we must support.

**Comment by dev-gaur** [comment]
yes, I myself didn't have much clarity into this. Thanks for correcting me.

**Comment by gorkem** [comment]
FYA @maxandersen

**Comment by maxandersen** [comment]
In recent version of quarkus we now have remote dev that does not require maven/Gradle on the remote side. 

Would that be something that we could make work with jkube and have it pick up Kubernetes yaml if users activated that in quarkus ?

**Comment by manusa** [comment]
Closing as superseded -- JKube's `remote-dev` module now provides remote development capabilities that overlap with the Quarkus-specific watcher use case described here.

If you have a specific Quarkus hot-reload scenario not covered by `remote-dev`, please open a new issue describing your use case.
